### DISCUSSION TITLE
Tracking recurring liabilities as a creditor

### DISCUSSION THREAD

**Discussion by eps90** [firefly-iii/firefly-iii#6719] [Firefly III]
Hello! I hope I didn't mess up the title :)
I'm looking for a way how I can track whether anyone owes me money, e.g. for subscriptions. Example - Spotify Family plan, where each of family members owes me $X every month or I am expecting $X * Y every Y months. As far as I understand liabilities accounts, I can expect that someone owes me some amount of money and keeps paying the debt. However, I would like it to have without any specified amount in total. Then I would be able to see that John did not pay me this month or still owes me money for 3 months. 

Is it even possible in Firefly? Maybe even with some reports? Thanks!

**Comment by taker218** [comment]
I would use reccuring transactions for that.

Example:
Liability Account: Spotify Family Plan
Owes me money: John

Transaction every month:
Amount x from "Spotify Family Plan" to "John"

When "John" pays the amount x to you, you can transfer this to "Spotify Family Plan" and it will be zeroed out again.

If "John" pays you for 6 month up front, you do the same. "Transfer" the money to the "Spotify Family Plan" and let the reccuring transactions deduct the budget from that account.

**Reply by eps90** [reply]
That makes sense. I will try this out and come back here later. Thanks!

**Reply by eps90** [reply]
Ok I played around with different scenarios and I find your solution @taker218 not perfect. Transfering an income directly into a liability account makes it invisible for asset account - like I've never received this money. To have anything on place, I need to clone this transaction (from John) and one of them goes to "Spotify Family Plan", the other to my account. But in that case, John's account is visible as he paid me twice the same amount. Moreover, cloning the transaction is not supported by Rules so I'd have to either do it manually or with API (is there a clone feature there?) with some webhook.

I also studied the documentation a bit and tried to connect your idea with the one presented in "Liabilities" description:
> Firefly III keeps track of the balance of the liability and the amount due.
>
> You need two transactions when your friend pays you back the money. If you do not do this, you'll never be able to use the money.
> 
> 1. A deposit into the liability from your friend (a revenue account).
> 1. A deposit from the liability into your own asset account.
>
> The first transaction is your friend paying your back. This will lower the "amount due" amount. It will also raise the balance, which means this money is now available to you. When you transfer the money from the liability to your own account, the balance is zero again but the "amount due" will not change.

That works even more weird. Also requires two transactions (John pays 2 times the same amount; no automation when importing transactions), and the liabilities show constantly increasing debt, while the "Amount due" is being zeroed. 

I think what I'm looking for is kind of "Bill" but with negative value (it's not me who pays bill but I'm expecting bills be paid to me)

Anyway, thanks a lot for your help! :muscle:

**Reply by JC5** [reply]
I know, I'm still looking for a nice way to make this work. Thanks for your input!
